contracts/StakingVe.sol

```diff
@@ -0,0 +1,305 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.6.12;
```

Comment:
We could use a newer version of solidity as this is a new contract with no dependencies on a specific version.

Reply:
Great, let's use 0.8.9 and have implicit safemath ftw